NewHaven/src/Setting.cpp

```diff
@@ -0,0 +1,141 @@
+        it->resourceTracker().score->insert(pair<ResourceTypes,std::uint_least16_t>(ResourceTypes::WHEAT,0));
+        it->resourceTracker().score->insert(pair<ResourceTypes,std::uint_least16_t>(ResourceTypes::STONE,0));
+        it->resourceTracker().score->insert(pair<ResourceTypes,std::uint_least16_t>(ResourceTypes::SHEEP,0));
+        it->resourceTracker().score->insert(pair<ResourceTypes,std::uint_least16_t>(ResourceTypes::WOOD,0));
+    }
+    cout << "GAME RESOURCE MARKERS SET SUCCESSFULLY" << endl;
+}
+
+int Setting::promptNumberPlayers() {
+    int number_of_players;
+    while ((cout << "Enter the number of players (2, 3 or 4): " && !(cin >> number_of_players))
+    || number_of_players < 2 || number_of_players >4) {
+        std::cin.clear(); //clear bad input flag
+        std::cin.ignore(std::numeric_limits<std::streamsize>::max(), '\n'); //discard input
+        std::cout << "Invalid input; please re-enter.\n";
+    }
+    return number_of_players;
+}
+
+inline void Setting::createHarvestDeck() {
+    cout << "CREATING HARVEST DECK!" << endl;
+    h_deck = new HarvestDeck();
+    cout << "HARVEST DECK CREATED SUCCESFULLY!" << endl;
+}
+
+inline void Setting::createBuildingDeck() {
+    cout << "CREATING BUILDING DECK!" << endl;
+    b_deck = new BuildingDeck();
+    cout << "BUILDING DECK CREATED SUCCESFULLY!" << endl;
+}
+
+inline Building* Setting::drawBuilding() {
+    cout << "DRAWING A BUILDING" << endl;
+    if(b_deck == nullptr)
+        return nullptr;
+    return b_deck->draw();
+}
+
+inline HarvestTile* Setting::drawHarvestTile() {
+    cout << "DRAWING A HARVEST TILE" << endl;
+    if(h_deck == nullptr)
+        return nullptr;
+    return h_deck->draw();
+}
+
+inline int Setting::getNumberPlayers() {
+    if(players== nullptr)
+        return 0;
+    return players->size();
+}
+
+bool Setting::initSetting() {
+    string files[3] = {"../config/GBMapConfig_0.config",
+                       "../config/GBMapConfig_1.config",
+                       "../config/GBMapConfig_2.config" };
+    int number_players = promptNumberPlayers();
+    switch(number_players){
+        case 2:
+            loadGameBoard(files[0]);break;
+        case 3:
+            loadGameBoard(files[1]);break;
+        case 4:
+            loadGameBoard(files[2]);break;
+        default:
+            loadGameBoard(files[0]);break;
+    }
+    setupPlayers(number_players);
+    resourceTracker();
+    createBuildingDeck();
+    createHarvestDeck();
+    for(std::vector<Player>::iterator it = players->begin() ; it != players->end(); ++it){
+        cout << "NEW PLAYER GETTING READY TO DRAW HIS/HER INTIAL SETUP!" << endl;
+        for(int i = 0; i < 5; i++)
+             it->drawBuilding(drawBuilding());
+        for(int j = 0; j < 2; j++)
+            it->drawHarvestTile(drawHarvestTile());
+        it->setShipmentTile(drawHarvestTile());
+
+        cout << "THE NEW PLAYER HAS FINISHED DRAWING THEIR HARVEST TILES AND BUILDINGS!" << endl;
+    }
```

Comment:
I think it would be better to have this done in the Hand default constructor.

Reply:
It would be, however, for Part 1 the teacher wants this to be a separate step. I'm already skipping one with the pool being auto filled up by the default constructor for the BuildingDeck. With how to the letter the TA have been i preferred putting that here since "theoretically" assigning the shipment tile and creation of the hand are two different task.